antefoyer/antefoyer.py

```diff
@@ -121,46 +129,69 @@ def ante_charges(molecule, charge_style, net_charge=0.0,
     with temporary_directory() as tmpdir:
         with temporary_cd(tmpdir):
             # Save the existing molecule to file
-            _write_pdb(molecule,'ante_in.pdb')
+            _write_pdb(molecule, "ante_in.pdb")
             # Call antechamber
-            command = ( 'antechamber -i ante_in.pdb -fi pdb '
-                         '-o ante_out.mol2 -fo mol2 ' +
-                         '-c ' + charge_style + ' ' +
-                         '-nc ' + str(net_charge) + ' ' +
-                         '-m ' + str(multiplicity) +  ' ' +
-                         '-s 2' )
-
-            proc = Popen(command, stdout=PIPE, stderr=PIPE,
-                         universal_newlines=True, shell=True)
+            command = (
+                "antechamber -i ante_in.pdb -fi pdb "
+                "-o ante_out.mol2 -fo mol2 "
+                + "-c "
+                + charge_style
+                + " "
+                + "-nc "
+                + str(net_charge)
+                + " "
+                + "-m "
+                + str(multiplicity)
+                + " "
+                + "-s 2"
+            )
+
+            proc = Popen(
+                command, stdout=PIPE, stderr=PIPE, universal_newlines=True, shell=True
+            )
 
             out, err = proc.communicate()
 
             # Error handling here
-            if 'Fatal Error' in err or proc.returncode != 0:
-                _antechamber_error(out,err,workdir)
+            if "Fatal Error" in err or proc.returncode != 0:
+                _antechamber_error(out, err, workdir)
 
             # Now read in the mol2 file with atomtyping
-            charges = pmd.load_file('ante_out.mol2',structure=True)
+            charges = pmd.load_file("ante_out.mol2", structure=True)
+
+            charge_list = [atom.charge for atom in charges]
+            if (net_charge - sum(charge_list)) > charge_tol:
```

Comment:
Do we need absolute values here to handle positive or negative net charges?

Reply:
Oops, forgot to push the absolute value.